File: app/src/main/java/chenmc/sms/transaction/SmsReceiver.java
package chenmc.sms.transaction;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.provider.Telephony;

/**
 * 接收短信广播的接收器
 * @author 明 明
 *         Created on 2017-2-20.
 */

public class SmsReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        switch (intent.getAction()) {
            case Telephony.Sms.Intents.SMS_RECEIVED_ACTION:
                // 打开一个服务进行相关处理
                intent.setClass(context, SmsReceiveService.class);
                context.startService(intent);
                break;
        }
    }
}

File: app/src/main/java/chenmc/sms/transaction/receiver/SmsReceiver.kt
package chenmc.sms.transaction.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.provider.Telephony
import chenmc.sms.transaction.service.SmsReceiveService

/**
 * 接收短信广播的接收器
 * @author 明 明
 * Created on 2017-2-20.
 */

class SmsReceiver : BroadcastReceiver() {
    
    override fun onReceive(context: Context, intent: Intent) {
        when (intent.action) {
            Telephony.Sms.Intents.SMS_RECEIVED_ACTION -> {
                // 打开一个服务进行相关处理
                intent.setClass(context, SmsReceiveService::class.java)
                context.startService(intent)
            }
        }
    }
}
